### PR TITLE
Added ingress support to the logstash chart 

### LINKED ISSUE #453: Ingress support in Logstash
The current Logstash helm chart lacks ingress support. I did find, in [a commit](https://github.com/elastic/helm-charts/commit/1768d4b24c4130566ff47017013bbb7ff8d1fde2) that it was initially added but removed soon after.

It states:

> Logstash HTTP API shouldn't be exposed externally as Logstash doesn't manage any authentication on this API.
> Also as Logstash doesn't support clustering, being able to access this API via a service doesn't make sense.

As far as I saw, the HTTP plugin does support some level of authentication and I really can't see how the lack of clustering support influences this decision. Do these conditions still hold?

### PR DESCRIPTION
**Pull request by jnbelo**
Added an ingress template to the logstash chart in order to enable external usage of the [http input plugin](https://www.elastic.co/guide/en/logstash/current/plugins-inputs-http.html) as well as exposing other http ports. Closes #453 .

- [x] Chart version *not* bumped (the versions are all bumped and released at the same time)
- [x] README.md updated with any new values or changes
- [x] Updated template tests in `${CHART}/tests/*.py` 
- [ ] Updated integration tests in `${CHART}/examples/*/test/goss.yaml`

### CODE REVIEW

**Review by ebuildy — DISMISSED** [review]

**Comment by ebuildy** [comment]
Excellent work!

**Comment by jmlrt** [comment]
jenkins test this please

File: logstash/tests/logstash_test.py [hunk 1]
```diff
@@ -876,3 +876,33 @@ def test_setting_fullnameOverride():
         ]
         == "logstash"
     )
+
```
Comment (jmlrt) [inline logstash/tests/logstash_test.py:879]: An empty line is missing here to make [Black](https://black.readthedocs.io/en/stable/) formatter happy in [elastic+helm-charts+pull-request+lint-python/484](https://devops-ci.elastic.co/job/elastic+helm-charts+pull-request+lint-python/484/label=docker&&virtual&&ubuntu-18.04/console).

Can you run `black <CHART>/tests/<CHART>_test.py`?

**Review by jmlrt — CHANGES_REQUESTED** [review]
Hi @jnbelo,
Thank you for submitting this PR 👍.
Some changes are required to make tests pass.
Otherwise the rest is LGTM.

File: logstash/values.yaml [hunk 2]
```diff
@@ -246,3 +246,13 @@ service: {}
 #      port: 8080
 #      protocol: TCP
 #      targetPort: 8080
+
+ingress:
+  enabled: true
```
[the remark this answers is not in the document]
Comment (jnbelo) [reply logstash/values.yaml]: Done

File: logstash/templates/ingress.yaml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+{{- if .Values.ingress.enabled -}}
+{{- $fullName := include "logstash.fullname" . -}}
+{{- $ingressPath := .Values.ingress.path -}}
```
[the remark this answers is not in the document]
Comment (jnbelo) [reply logstash/templates/ingress.yaml]: Forgot to remove this from my initial implementation! Done.

File: logstash/tests/logstash_test.py (hunk 1 above)
Comment (jnbelo) [reply logstash/tests/logstash_test.py:878]: Done.

**Comment by jnbelo** [comment]
@ebuildy @jmlrt  Thanks for the review! The comments were addressed.

**Review by jmlrt — APPROVED** [review]
LGTM⛴

**Comment by jmlrt** [comment]
jenkins test this please

**Comment by jmlrt** [comment]
backported to `6.8`, `7.9` and `7.x` branches

**Pull request merged**